quint/src/apalache.ts

```diff
@@ -0,0 +1,430 @@
+/* ----------------------------------------------------------------------------------
+ * Copyright 2023 Informal Systems
+ * Licensed under the Apache License, Version 2.0.
+ * See LICENSE in the project root for license information.
+ * --------------------------------------------------------------------------------- */
+
+/**
+ * Interface to Apalache
+ *
+ * This functionality is enabled by managing and interacting with the Apalache
+ * server.
+ *
+ * @author Shon Feder
+ *
+ * @module
+ */
+
+import { Either, chain, left, right } from '@sweet-monads/either'
+import { ErrorMessage } from './ErrorMessage'
+import path from 'path'
+import fs from 'fs'
+import os from 'os'
+// TODO: used by GitHub api approach: https://github.com/informalsystems/quint/issues/1124
+// import semver from 'semver'
+import { pipeline } from 'stream/promises'
+import child_process from 'child_process'
+import * as tar from 'tar'
+import * as grpc from '@grpc/grpc-js'
+import * as proto from '@grpc/proto-loader'
+import * as protobufDescriptor from 'protobufjs/ext/descriptor'
+import { setTimeout } from 'timers/promises'
+import { promisify } from 'util'
+import { ItfTrace } from './itf'
+import { verbosity } from './verbosity'
+// TODO: used by GitHub api approach: https://github.com/informalsystems/quint/issues/1124
+// import { request as octokitRequest } from '@octokit/request'
+
+import type { Buffer } from 'buffer'
+import type { PackageDefinition as ProtoPackageDefinition } from '@grpc/proto-loader'
+
+const APALACHE_SERVER_URI = 'localhost:8822'
+const APALACHE_VERSION_TAG = '0.44.7'
+// TODO: used by GitHub api approach: https://github.com/informalsystems/quint/issues/1124
+// const APALACHE_TGZ = 'apalache.tgz'
+
+function quintConfigDir(): string {
+  return path.join(os.homedir(), '.quint')
+}
+
+function apalacheDistDir(): string {
+  return path.join(quintConfigDir(), `apalache-dist-${APALACHE_VERSION_TAG}`)
+}
+
+// The structure used to report errors
+type ApalacheError = {
+  explanation: string
+  errors: ErrorMessage[]
+  traces?: ItfTrace[]
+}
+
+export type ApalacheResult<T> = Either<ApalacheError, T>
```

Comment:
The only change here aside from moving stuff was to rename these types (that used to have `Verify` instead of `Apalache`).